CMakeLists.txt

```diff
@@ -1,18 +1,47 @@
+#Set the minimum version of CMake. I have set this to 3.1 as it supports 
+# CXX 11 and 14 out of the box. CXX 17 can be used by setting the CXX Flags
 cmake_minimum_required (VERSION 3.1)
 
+set(MAJOR_VERSION 1)
+set(MINOR_VERSION 0)
+set(PATCH_VERSION 4)
+
+# Set the project name
 project (build_system_playground)
 
 
-#set(CMAKE_CXX_STANDARD 17)
+#compiler options setup
+#set(CMAKE_CXX_STANDARD 14)
 option(MULTITHREAD_SUPPORT_ENABLED "Multithreaded Support Enabled" TRUE)
-SET( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -std=c++1z" )
-add_library(build_system_playground SHARED src/build_system_playground/main.cpp)
+set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -std=c++1z" )
```

Comment:
Yeah. but this is more explicit as to adding it to the flags. CMake 3.8.0 adds c++ 17 support to the CMAKE_CXX_STANDARD macro. It is not released yet though. :(